Week4/app/src/main/java/com/isabelledionisius/mymovieapp/MainActivity.kt

```diff
@@ -0,0 +1,58 @@
+package com.isabelledionisius.mymovieapp
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+
+/** Main activity of the list page
+ *
+ *
+ */
+
+class MainActivity : AppCompatActivity(),MovieAdapter.MovieListClickListener {
+
+    private lateinit var movieListRecyclerView: RecyclerView
+    private var movieDataManager = MovieAdapter(this)
+
+    companion object {
+        const val INTENT_LIST_KEY = "list"
+    }
+
+    /**
+     * Creating the adapter and assigning the values to it
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        movieListRecyclerView = findViewById(R.id.recycler_view)
+        movieListRecyclerView.layoutManager = LinearLayoutManager(this)
+        // movieDataManager.getList()
+        // I think the error is here, I know that I need to pass in the list with my data
+        movieListRecyclerView.adapter = MovieAdapter(this)
+        //showDetailMovieItems(movieDataManager.getList())
+    }
+
+    /**
+     * This method will open the new activity passing the tasklist objects to it
+     * we are passing in the context which is the activity to the Intent => "this"
+     * then we are addressing the new class
+     * -- check the parameter if that is the correct one: in the examle: list:TaskList
+     */
+    private fun showDetailMovieItems(list: Movie) {
+        val detailMovieItem = Intent(this, DetailActivity::class.java)
```

Comment:
try the `getIntent()` pattern, by adding a companion object to the Details activity, and then building the intent in that static function.

example:

```kotlin
class DetailActivity ... {

  companion object {
    fun getIntent(context: Context, params...) = Intent(context, DetailActivity::class.java).apply {
      putExtra(params...)
    }
  }
}
```